proto/streamlit/proto/DeckGlJsonChart.proto

```diff
@@ -28,9 +28,12 @@ message DeckGlJsonChart {
   // If True, will overwrite the chart width spec to fit to container.
   bool use_container_width = 4;
 
-  // ID
+  // ID, required for selection events.
   string id = 5;
 
   // The user-configured Mapbox token. If empty, the token id fetched from https://data.streamlit.io/tokens.json
   string mapbox_token = 6;
+
+  // If True, treat this instance as a Widget
+  bool has_selection = 7;
```

Comment:
Are we sure selection is just flag to turn on/off? Just looking at other examples, we have the concept of `SelectionMode` which for maps can be single or multiselect. Looking at the spec, seems like we punted on the decision, but perhaps we should make this an enum with only one option `multiselect` and add `single` if needed?

Reply:
I also added a few thoughts about selection mode in the spec [here](https://www.notion.so/snowflake-corp/Spec-a9163e158e754848a484093fc76bf550?pvs=4#1027170bb41680139808f8553c88738f). Even if we don't add it right now, I agree that it might make sense to directly make this an enum. 

In theory, we have the commitment to keep our protos backwards compatible. This is a lot easier to do for future updates if we keep this property extendable.

Reply:
Updated this to have an explicit `selection_mode` in its proto. Please take a look!